applications/signals.py

```diff
@@ -1,13 +1,59 @@
-from django.db.models.signals import post_save
+import os
+
+from django.db.models.signals import post_save, post_delete, pre_save
 from django.dispatch import receiver
+from django.forms import model_to_dict
 
 from applications import models
 
 
 # Delete DraftApplication when application submitted
-@receiver(post_save, sender=models.Application)
+@receiver(post_save, sender=models.HackerApplication)
+@receiver(post_save, sender=models.MentorApplication)
+@receiver(post_save, sender=models.VolunteerApplication)
 def clean_draft_application(sender, instance, created, *args, **kwargs):
     if not created:
         return None
     # Delete draft as its no longer needed
     models.DraftApplication.objects.filter(user=instance.user).delete()
+
+
+# Create DraftApplication when application deleted
+@receiver(post_delete, sender=models.HackerApplication)
+@receiver(post_delete, sender=models.MentorApplication)
+@receiver(post_delete, sender=models.VolunteerApplication)
+def create_draft_application(sender, instance, *args, **kwargs):
+    dict = model_to_dict(instance)
+    for key in ['user', 'invited_by', 'submission_date', 'status_update_date', 'status', 'resume']:
+        dict.pop(key, None)
+    d = models.DraftApplication()
+    d.user = instance.user
+    d.save_dict(dict)
+    d.save()
```

Comment:
Why is this?

Reply:
Imagine a hacker wants to be a mentor or otherwise or to volunteer. There are fields that will be the same, so if we save that data as draft application, later the user won't have to put it again

Reply:
Also the max amount of data that can be in db is N (number of users), users have of application or a draft application, cannot be both. There was an issue that made signals didn't work so the amount of data right now is 2*N because every user that have an application also have a draft application. This is solved in this pull request.